core/src/main/java/bisq/core/trade/TradeUtil.java

```diff
@@ -108,4 +118,90 @@ public TradeUtil(BtcWalletService btcWalletService, KeyRing keyRing) {
 
         return new Tuple2<>(multiSigAddress.getAddressString(), payoutAddress);
     }
+
+    public long getRemainingTradeDuration(Trade trade) {
+        return trade.getMaxTradePeriodDate() != null
+                ? trade.getMaxTradePeriodDate().getTime() - new Date().getTime()
+                : getMaxTradePeriod(trade);
+    }
+
+    public long getMaxTradePeriod(Trade trade) {
+        return trade.getOffer() != null
+                ? trade.getOffer().getPaymentMethod().getMaxTradePeriod()
+                : 0;
+    }
+
+    public double getRemainingTradeDurationAsPercentage(Trade trade) {
+        long maxPeriod = getMaxTradePeriod(trade);
+        long remaining = getRemainingTradeDuration(trade);
+        if (maxPeriod != 0) {
+            return 1 - (double) remaining / (double) maxPeriod;
+        } else
+            return 0;
+    }
+
+    public String getRemainingTradeDurationAsWords(Trade trade) {
+        return formatDurationAsWords(Math.max(0, getRemainingTradeDuration(trade)));
+    }
+
+    @Nullable
+    public Date getHalfTradePeriodDate(Trade trade) {
+        return trade != null ? trade.getHalfTradePeriodDate() : null;
+    }
+
+    public Date getDateForOpenDispute(Trade trade) {
+        return new Date(new Date().getTime() + getRemainingTradeDuration(trade));
+    }
+
+    public String getMarketDescription(Trade trade) {
+        if (trade == null)
+            return "";
+
+        checkNotNull(trade.getOffer());
+        checkNotNull(trade.getOffer().getCurrencyCode());
+        return getCurrencyPair(trade.getOffer().getCurrencyCode());
+    }
+
+    public String getPaymentMethodNameWithCountryCode(Trade trade) {
+        String paymentMethodDescription = "";
+        if (trade != null) {
```

Comment:
Prefer to return early, ie
```
if (trade == null)
    return;

Offer offer = checkNotNull(trade.getOffer());
checkNotNull(offer.getPaymentMethod());
return offer.getPaymentMethodNameWithCountryCode();
```

Reply:
Resolved in PR https://github.com/bisq-network/bisq/pull/4731, commit https://github.com/bisq-network/bisq/pull/4731/commits/f764e9feb07f1cae45309b6271ef84862ee7d8b4.